src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js

```diff
@@ -0,0 +1,196 @@
+                var itemData      = getItemData(typeLowerCase);
+
+                if (!allowReplace && itemData)
+                    throw new Error(`Failed to execute 'add' on 'DataTransferItemList': An item already exists for type '${typeLowerCase}'.`);
+
+                if (itemData)
+                    removeItem(typeLowerCase);
+
+                newItem = new DataTransferItem(DATA_TRANSFER_ITEM_KIND.string, processFormat(type), data);
+            }
+            else
+                newItem = new DataTransferItem(DATA_TRANSFER_ITEM_KIND.file, null, data);
+
+            items.push(newItem);
+            itemsData.push(data);
+            updateIndexes();
+
+            return newItem;
+        };
+
+        // Internal API
+        this.getAndHideInternalMethods = () => {
+            var res = { getTypes, getItemData, removeItem, addItem };
+
+            this.getAndHideInternalMethods = void 0;
+
+            return res;
+        };
+
+        // API
+        Object.defineProperty(this, 'length', {
+            enumerable: true,
+
+            get: () => items.length
+        });
+
+        Object.defineProperty(this, 'remove', {
+            configurable: true,
+            enumerable:   true,
+
+            get: () => {
+                return function (idx) {
+                    if (dataStore.mode !== DATA_STORE_MODE.readwrite)
+                        return;
+
+                    items.splice(idx, 1);
+                    itemsData.splice(idx, 1);
+                    updateIndexes();
+                };
+            }
+        });
+
+        Object.defineProperty(this, 'clear', {
+            configurable: true,
+            enumerable:   true,
+
+            get: () => {
+                return function () {
+                    if (dataStore.mode !== DATA_STORE_MODE.readwrite)
+                        return;
+
+                    items     = [];
+                    itemsData = [];
+                    updateIndexes();
+                };
+            }
+        });
+
+        Object.defineProperty(this, 'add', {
+            configurable: true,
+            enumerable:   true,
+
+            get: () => {
+                return function (data, type) {
+                    if (!arguments.length)
+                        throw new Error("Failed to execute 'add' on 'DataTransferItemList': 1 argument required, but only 0 present.");
+
+                    if (arguments.length === 1 && typeof data === 'string')
+                        throw new Error("Failed to execute 'add' on 'DataTransferItemList': parameter 1 is not of type 'File'.");
+
+                    if (dataStore.mode !== DATA_STORE_MODE.readwrite)
+                        return void 0;
+
+                    return addItem(data, type, false);
+                };
+            }
+        });
+    }
+}
+
+if (window.DataTransferItemList)
+    DataTransferItemList.prototype = window.DataTransferItemList.prototype;
```

Comment:
Save to `DataTransferItemList` to `nativeMethods` and use `nativeMethods.prototype.DataTransferItemList`